linera-storage/src/db_storage.rs

```diff
@@ -474,13 +474,14 @@ where
     }
 
     async fn missing_blobs(&self, blob_ids: Vec<BlobId>) -> Result<Vec<BlobId>, ViewError> {
-        let mut keys = Vec::new();
+        let mut keys = Vec::with_capacity(blob_ids.len());
         for blob_id in blob_ids.clone() {
```

Comment:
(Not your fault, but: No need to allocate a new vector.)

```suggestion
        for blob_id in blob_ids.iter().cloned() {
```